Mage.Sets/src/mage/cards/m/MutatedCultist.java

```diff
@@ -0,0 +1,166 @@
+import mage.util.CardUtil;
+import mage.watchers.common.CastSpellLastTurnWatcher;
+
+/**
+ *
+ * @author grimreap124
+ */
+public final class MutatedCultist extends CardImpl {
+
+    public MutatedCultist(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{2}{B}");
+        
+        this.subtype.add(SubType.ELDRAZI);
+        this.subtype.add(SubType.HORROR);
+        this.power = new MageInt(1);
+        this.toughness = new MageInt(3);
+
+        // Devoid
+        this.addAbility(new DevoidAbility(this.color));
+
+        // When you cast this spell, remove all counters from up to one target permanent or opponent. The next spell you cast this turn costs {1} less to cast for each counter removed this way.
+        Ability ability = new CastSourceTriggeredAbility(new MutatedCultistEffect());
+        ability.addTarget(new TargetPermanentOrPlayer());
+        this.addAbility(ability);
+        // Deathtouch
+        this.addAbility(DeathtouchAbility.getInstance());
+
+    }
+
+    private MutatedCultist(final MutatedCultist card) {
+        super(card);
+    }
+
+    @Override
+    public MutatedCultist copy() {
+        return new MutatedCultist(this);
+    }
+}
+
+class MutatedCultistEffect extends OneShotEffect {
+
+    private static final FilterPermanent filter = new FilterPermanent("permanent with a counter");
+
+    static {
+        filter.add(CounterAnyPredicate.instance);
+    }
+
+    MutatedCultistEffect() {
+        super(Outcome.BoostCreature);
+        this.staticText = "remove all counters from up to one target permanent or opponent. The next spell you cast this turn costs {1} less to cast for each counter removed this way.";
+    }
+
+    private MutatedCultistEffect(final MutatedCultistEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public MutatedCultistEffect copy() {
+        return new MutatedCultistEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        int countersRemoved = 0;
+        UUID targetId = source.getTargets().getFirstTarget();
+        if (targetId == null) {
+            return false;
+        }
+        Player targetedPlayer = game.getPlayer(targetId);
+        if (targetedPlayer != null) {
+            countersRemoved = targetedPlayer.loseAllCounters(source, game);
+        }
+        Permanent targetPermanent = game.getPermanent(targetId);
+        if (targetPermanent != null) {
+            countersRemoved = targetPermanent.removeAllCounters(source, game);
+        }
+        if (countersRemoved > 0) {
+            game.informPlayers("Removed " + countersRemoved);
+            game.addEffect(new MutatedCultistSpellsCostReductionEffect(countersRemoved).setDuration(Duration.OneUse), source);
+            return true;
+        }
+        return false;
+    }
+}
+
+class MutatedCultistSpellsCostReductionEffect extends CostModificationEffectImpl {
```

Comment:
this really should be broken out into a common class, but that's a separate scope

Reply:
I could find only 1 other card that had similar code (that I took the base of this from) and thought about making it a class that could handle both but wasn't confident I could get it right